lib/std/os.zig

```diff
@@ -2144,14 +2151,16 @@ pub fn mkdiratW(dir_fd: fd_t, sub_path_w: [*:0]const u16, mode: u32) MakeDirErro
 /// Create a directory.
 /// `mode` is ignored on Windows.
 pub fn mkdir(dir_path: []const u8, mode: u32) MakeDirError!void {
+    if (builtin.os.tag == .wasi) {
+        @compileError("mkdir is not supported in WASI; use mkdirat instead");
+    }
     if (builtin.os.tag == .windows) {
         const sub_dir_handle = try windows.CreateDirectory(null, dir_path, null);
         windows.CloseHandle(sub_dir_handle);
         return;
-    } else {
-        const dir_path_c = try toPosixPath(dir_path);
```

Comment:
why move all these out of the `else` branches?

Reply:
Oh oops, they must have slipped in from my experimentation with different indentations re readability. I'll fix it right away, thanks for spotting that @daurnimator!

Reply:
OK, I've reverted to old formatting plus added `@compileError` in `fstat` and `fstatat` when targeting Windows. Done in e401a52.